### DISCUSSION TITLE
Definitions - text, image, content

### DISCUSSION THREAD

**Discussion by alastc** [w3c/wcag3#140] [General]
There are quite a few definitions already used by WCAG 3 and it's guidelines that are either the drawn straight from WCAG 2, or not defined yet.

As a starting point, there are a few basic and related definitions it would be useful to work on, with initial suggestions:

- Text: human language[link] encoded in a directly machine-readable way. Example: UTF-8 text rendered by a browser.
- Image: A visual representation of something encoded in a format for display on a screen.
- Image of text: text[link to text] that has been rendered as an image[link to image].
- Content: information and sensory experience to be communicated to the user.

The update to ["text" from WCAG 2](https://www.w3.org/TR/WCAG22/#dfn-text) is because (IMHO) an image of text with alt-text appears to meet that definition.
The ["content" definition from WCAG 2](https://www.w3.org/TR/WCAG22/#dfn-text) assumes a user-agent, which should not be an assumption in WCAG 3.

Related definitions from WCAG 2 with no suggested update:

- human language: language that is spoken, written or signed (through visual or tactile means) to communicate with humans.
- user interface component: a part of the content that is perceived by users as a single control for a distinct function 

If you have suggestions for these definitions, please comment below. 

Two related questions for discussion:
1. Are there any other issues you've spotted with these definitions when using WCAG 2 (that we can solve here).
2. Should we separate controls (user-interface-components) from content further? E.g. try to exclude controls from content.

**Comment by gundulaniemann** [comment]
In order to understand the impact of the second question I would like to understand the background of the definition of "content". How is the term used? What is supposed to be expressed by the term?
the 'sensory experience' suggests that the interaction aspects are part of it, which means controls are part of it.

**Comment by dbjorge** [comment]
> Should we separate controls (user-interface-components) from content further? E.g. try to exclude controls from content.

I'm all for creating clear definitions for controls vs content, but I'm hesitant to make them non-overlapping; it's very common in practice for sites to use components that are ambiguous or overlapping about controls vs content - a landing page might present content in "cards" that are clickable links, a text editor's main panel is a big control but also contains most of the content being displayed, a chat app might let you click on messages to edit them/etc...

**Comment by bruce-usab** [comment]
1. Nice job with these four updated terms!  Subtle but important improvements.
2. I think UIC has to be a subset of content (i.e., "a part of the content") but that might be my own lack of creativity.  The idea might merit exploring.

**Comment by alastc** [comment]
Trying to bullet proof my own suggestion:
> Text: human language[link] encoded in a directly machine-readable way. Example: UTF-8 text rendered by a browser.

Human language can be (according to the definition) audio or signed, as well as written. So I think we need to include that text is in the written form, e.g.:

**Text**: written human language[link] encoded in a directly machine-readable way. Example: UTF-8 text rendered by a browser.

**Reply by dbjorge** [reply]
> The update to ["text" from WCAG 2](https://www.w3.org/TR/WCAG22/#dfn-text) is because (IMHO) an image of text with alt-text appears to meet that definition.

What actual problems does that cause? Even if there are some, why do we have to throw out other parts of the existing definition that aren't related to solve it? (eg, the terminology and examples around "sequence of characters", which I think is basically already a more precise way of solving what you're looking at with re-adding the word "written")

> encoded in a directly machine-readable way

I'd stay away from this terminology unless you define it in turn; I would argue that a jpg file or a zip file are encoded in a "directly machine-readable way".

> Example: UTF-8 text rendered by a browser.

I know this is just an example rather than the definition itself, but it suggests that the text might need to be "rendered" to count, which we probably don't want (text alternatives are text but often aren't "rendered")

**Reply by alastc** [reply]
Hi Dan, quite a few guidelines are going to need a concept of user-manipulable text. An image with alt-text prevents that (or at least makes it much harder to do). For example, a clear-language guideline might have a basic requirement of user-manipulable text (allowing for user-agents to do the work), and at a higher level have author requirements.

For the text definition, or something building on top of it, we need a concept of "real" text. 

How about:
"a sequence of characters represented visually and expressing something in human language.  Example: UTF-8 text rendered by a browser."

**Reply by alastc** [reply]
Hmm, I don't think that works, I'm really looking for something that scopes in rended text, and scopes out image-based text with an alternative.

**Comment by alastc** [comment]
### Summary 3rd Dec 2024

General agreement with the definitions, one proposed tweak to "text".

> Text: written human language[link] encoded in a directly machine-readable way. Example: UTF-8 text rendered by a browser.

Several comments that controls are likely to be included within "content" in most people's minds. 

Possible alternative for discussion:

- **user Interface**: information and sensory experience to be communicated to the user, which may also allow for action or input from the user.
- **user interface control**: a part of the _user interface_ that is perceived by users as a single control for a distinct function.

That provides a term for the holistic "thing people use", and the defines an obvious sub-set for things that take input.

**Reply by scottaohara** [reply]
i would much prefer these terms, than the general "content" term.  these proposals make far more sense to me.

**Reply by alastc** [reply]
GN suggestion: "action or input" from the user.
